packages/execution-environments/src/common/commands.ts

```diff
@@ -26,14 +25,18 @@ export type CommandMethodsMapping = {
  * @param request - The request object.
  * @returns The formatted arguments.
  */
-function getHandlerArguments(
-  origin: Origin,
+export function getHandlerArguments(
+  origin: string,
   handler: HandlerType,
-  request: JsonRpcRequest<unknown[] | { [key: string]: unknown }>,
+  request: JsonRpcRequestWithoutId,
 ): InvokeSnapArgs {
+  // `request` is already validated by the time this function is called.
```

Comment:
Except for `onTransaction`? Or why is the new assert needed

Reply:
We validate the `request`, where `params` can be any valid JSON-RPC params.